### ISSUE TITLE
Multiline CSS selectors do not work correctly with ssr

### ISSUE THREAD

**Issue by maoberlehner** [nuxt/nuxt.js#2110]
I use a lot of CSS classes and I like long, descriptive class names, thats why I sometimes end up with multi line class names:

```html
<div
  class="
    o-grid__item
    o-vertical-spacing
    o-vertical-spacing--l
    u-width-12/12
    u-width-6/12@m">
</div>
```

It seems that the server renderer can't handle those new line separated classes:

<img width="808" alt="bildschirmfoto 2017-11-14 um 18 08 50" src="https://user-images.githubusercontent.com/6883314/32793758-1ccae744-c967-11e7-9e2f-0d1b1c330beb.png">

As soon as I trigger client side rendering, everything looks fine again:

<img width="273" alt="bildschirmfoto 2017-11-14 um 18 11 10" src="https://user-images.githubusercontent.com/6883314/32793793-3aa40dc2-c967-11e7-8430-02b1478ab8f5.png">

Is there something I can do about this or is this a bug in Nuxt or the Vue server renderer?

**Comment by clarkdo** [comment]
The issue should come from `vue-template-compiler`, it won't `escape` class if there is no class binding on div except the fisrt tag of template.

For a quick fix, you can add a `:class="{}"` on the div:

```html
<div
       :class="{}
        class="
          o-grid__item
          o-vertical-spacing
          o-vertical-spacing--l
          u-width-12/12
          u-width-6/12@m">
```

**Comment by maoberlehner** [comment]
@clarkdo thanks for your quick response. I changed `class="..."` to `class="{...}` as a temporary workaround.

**Comment by felixbuenemann** [comment]
@clarkdo, please re-open, the workaround cannot be used with responsive images, were multi line is often used for sizes, srcset, etc.

**Comment by felixbuenemann** [comment]
I was able to reproduce the issue just with `vue-template-compiler`. The problem is that the compiler is double escaping newlines in attributes when it is creating ssrNode string templates:

```javascript
template = ("<div>\n<textarea id=\"foo,\nbar\">foo\nbar</textarea>\n</div>"
result = compiler.ssrCompile(template)
console.log(result.render)
// output: with(this){return _c('div',[_ssrNode("<textarea id=\"foo,\\nbar\">foo\nbar</textarea>")])}
```

**Comment by felixbuenemann** [comment]
I will report the issue upstream and link back here.

**Comment by felixbuenemann** [comment]
I opened https://github.com/vuejs/vue/issues/7223

**Comment by clarkdo** [comment]
@felixbuenemann 
As my above comments, it's a issue related to `class` in html tag.

I have tried below, it works fine with multiple lines, can you provider an example? 

```html
<img
      sizes="(min-width: 400px) 80vw, 100vw"
      secret="examples/images/small.jpg 375w,
              examples/images/big.jpg 1500w"/>
```

**Comment by clarkdo** [comment]
@felixbuenemann Thanks for your effort. 😸

**Comment by felixbuenemann** [comment]
Make sure there is another tag around `<img/>` or ssr will put secret in the attrs property instead of inside _ssrNode.

Here is a better example to show:

```javascript
compiler = require('vue-template-compiler')
template = "<div id=\"foo,\nbar\">\n<textarea id=\"foo,\nbar\">foo\nbar</textarea>\n</div>"
result = compiler.ssrCompile(template)
console.log(result.render)
// output: with(this){return _c('div',{attrs:{"id":"foo,\nbar"}},[_ssrNode("<textarea id=\"foo,\\nbar\">foo\nbar</textarea>")])}
```

As you can see it is not double escaped for the "id" put into "attrs" prop.

**Comment by felixbuenemann** [comment]
This is all part of the "optimizing compiler" that renders some stuff to simple strings in SSR mode instead of more complex code in non-ssr mode, which is why the behavior differs between SSR and rehydration.

I think this bug is also why we are seeing mismatch errors in rehydration, because server side markup differs due to the extra "\n" strings instead of actual newlines.

**Comment by felixbuenemann** [comment]
I proposed a fix for the issue in PR https://github.com/vuejs/vue/pull/7224.

**Comment by felixbuenemann** [comment]
My fix is included in vue-template-compiler v2.5.10.

@clarkdo Could you bump dependencies on dev branch to Vue.js 2.5.10?

**Comment by clarkdo** [comment]
@felixbuenemann  Sure
